### PR TITLE
feat/node: Update protocol parameters

### LINKED ISSUE #1134: Adapt ProtocolParameters - Nodejs
https://github.com/iotaledger/tips-draft/pull/25

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Updates the params

## Links to any relevant issues

Fixes [1134](https://github.com/iotaledger/iota-sdk/issues/1134)

### CODE REVIEW

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts [hunk 1]
```diff
@@ -169,42 +158,80 @@ export interface WorkScoreStructure {
...
 }
 
+/**
+ * Mana Structure defines the parameters used by mana calculation.
+ */
+export interface ManaStructure {
+    /**
+     * The number of bits used to represent Mana expressed as an exponent of 2.
+     */
+    bitsExponent: number;
```
Comment (Thoralf-M) [inline bindings/nodejs/lib/types/models/info/node-info-protocol.ts]:
```suggestion
    bitsCount: number;
```

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts [hunk 2]
```diff
@@ -169,42 +158,80 @@ export interface WorkScoreStructure {
...
 }
 
+/**
+ * Mana Structure defines the parameters used by mana calculation.
+ */
+export interface ManaStructure {
+    /**
+     * The number of bits used to represent Mana.
+     */
+     bitsCount: number;
```
Comment (Thoralf-M) [inline bindings/nodejs/lib/types/models/info/node-info-protocol.ts]: Remove the extra space before :trollface:

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts [hunk 3]
```diff
@@ -169,42 +158,80 @@ export interface WorkScoreStructure {
...
+     */
+    generationRate: number;
+    /**
+     * The scaling of ManaGenerationRate expressed as an exponent of 2.
+     */
+    generationRateExponent: number;
+    /**
+     * A lookup table of epoch index diff to mana decay factor (slice index 0 = 1 epoch).
+     */
+    decayFactors: number[];
```
Comment (abdulmth) [inline bindings/nodejs/lib/types/models/info/node-info-protocol.ts:180]: Maybe using `Uint16Array` could be worthwhile? but I'm not sure if that can cause other problems elsewhere.

**Review by abdulmth — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

File: bindings/nodejs/lib/types/models/info/node-info-protocol.ts (hunk 3 above)
Comment (kwek20) [reply bindings/nodejs/lib/types/models/info/node-info-protocol.ts:180]: yea that sounds scary :D

**Pull request merged**